### PR TITLE
Making gh-bot to respond to All Checks Have Passed.

### PR DESCRIPTION
**Pull request by arooshap**

### CODE REVIEW

**Comment by arooshap** [comment]
@smuzaffar could you please review this?

File: dmwmbld/gh-bot [hunk 1]
```diff
@@ -26,13 +26,23 @@ def generate_index_page(pr, status, cfgdir, statedir):
 # Test builds for pull requests against an specific branch in the gh repo.
 # Update the pull req status according to overall result of tests on all
 # the architectures.
+def check_context(gh, pull_request, owner, repo, branch):
+    statuses_url = pull_request['statuses_url']
+    # You can use the 'statuses_url' to fetch and check the context or status of the pull request
+    statuses = gh.get(statuses_url)
```
Comment (smuzaffar) [inline dmwmbld/gh-bot:32]: @arooshap , I do not see `get` fucntion in https://github.com/dmwm/deployment/blob/633ac3951fc18d7a487418eb065953d35b00b6cf/dmwmbld/github.py ? May be you want to call `GetStatus` ?

File: dmwmbld/gh-bot [hunk 2]
```diff
@@ -26,13 +26,23 @@ def generate_index_page(pr, status, cfgdir, statedir):
...
+    # Check if there's a status with context equal to "cms/bot"
+    return any(status['state'] == 'success' and status.get('context') == 'cms/bot' for status in statuses)
 def pull_req_tester(gh, owner, repo, branch, arch, all_archs, statedir, logdir, logurl, cfgdir, authfile):
   # get the relevant pull requests
   valid_comp_users = json.load(open(usersfile))
   PRs = [i for i in gh.GetPullRequests(owner, repo) \
            if (((i['base']['label'] == owner + ':' + branch) and
-           (i['user']['login'].encode('ascii', 'ignore') in valid_comp_users)) or
-           (any(label['name'] == 'cmsbot/test' for label in i.get_labels())))]
+           (i['user']['login'].encode('ascii', 'ignore') in valid_comp_users)) or (check_context(gh, i, owner, repo, branch))]
```
Comment (smuzaffar) [inline dmwmbld/gh-bot:41]: @arooshap , with current logic your bot will start testing all PRs which have 'cmsbot/test' status (even if PR is open for non `comp_gcc630` branch). I would suggest to change this to 
```
if ((<correct-branch>) and ((<valid_comp_user>) or (<pr-has-cmsbot/test status>)))
```
i.e. 
- PR should have a `correct-branch` and 
- either PR is opened by on of  `valid_comp_users`  OR PR has `cmsbot/test` status

File: dmwmbld/gh-bot (hunk 1 above)
Comment (arooshap) [reply dmwmbld/gh-bot:32]: This was my test script:
```
import requests

# Define the repository and pull request number
owner = 'cms-sw'
repo = 'cmsdist'
pull_request_number = 8757

# Create the API URL to get PR information
pr_url = f'https://api.github.com/repos/{owner}/{repo}/pulls/{pull_request_number}'


pr_response = requests.get(pr_url)

if pr_response.status_code == 200:
    pr_data = pr_response.json()

    # Extract the statuses URL from the PR data
    statuses_url = pr_data['statuses_url']

    # Make the API request to get the statuses for the PR
    statuses_response = requests.get(statuses_url)

    if statuses_response.status_code == 200:
        statuses_data = statuses_response.json()
        if statuses_data:
            context = statuses_data[0]['context']
            print(f"Context of the pull request: {context}")
        else:
            print('No status checks found for the pull request.')
    else:
        print(f'Error retrieving pull request status information: {statuses_response.status_code}')
else:
    print(f'Error retrieving pull request information: {pr_response.status_code}')

```

When I run this, I get the following:
```
[apervaiz@lxplus812 ~]$ python3 python456.py 
Context of the pull request: cmsbot/test

```

File: dmwmbld/gh-bot [hunk 3]
```diff
@@ -26,13 +26,23 @@ def generate_index_page(pr, status, cfgdir, statedir):
 # Test builds for pull requests against an specific branch in the gh repo.
 # Update the pull req status according to overall result of tests on all
 # the architectures.
+def check_context(gh, pull_request, owner, repo, branch):
+    statuses_url = pull_request['statuses_url']
+    # You can use the 'statuses_url' to fetch and check the context or status of the pull request
+    statuses = gh.get(statuses_url)
+    
+    # Check if there's a status with context equal to "cms/bot"
+    return any(status['state'] == 'success' and status.get('context') == 'cms/bot' for status in statuses)
```
Comment (smuzaffar) [inline dmwmbld/gh-bot:35]: context should be `cmsbot/test`

File: dmwmbld/gh-bot [hunk 4]
```diff
@@ -32,7 +32,7 @@ def pull_req_tester(gh, owner, repo, branch, arch, all_archs, statedir, logdir,
   PRs = [i for i in gh.GetPullRequests(owner, repo) \
            if (((i['base']['label'] == owner + ':' + branch) and
            (i['user']['login'].encode('ascii', 'ignore') in valid_comp_users)) or
-           (any(label['name'] == 'cmsbot/test' for label in i.get_labels())))]
+           (i.get('mergeable_state', '') == 'clean'))]
```
[the remark this answers is not in the document]
Comment (arooshap) [reply dmwmbld/gh-bot]: @smuzaffar I was using `requests` library in my test script, sorry for the overlook.

File: dmwmbld/gh-bot (hunk 1 above)
Comment (arooshap) [reply dmwmbld/gh-bot]: @smuzaffar I was using `requests` library in my test script, sorry for the overlook.

File: dmwmbld/gh-bot (hunk 2 above)
Comment (arooshap) [reply dmwmbld/gh-bot]: @smuzaffar I changed it to
```
  PRs = [i for i in gh.GetPullRequests(owner, repo) \
        if (i['base']['ref'] == 'comp_gcc630' and ((i['user']['login'].encode('ascii', 'ignore') in valid_comp_users) or check_context(gh, i, owner, repo, branch))]
```

Does it suffice?

Comment (smuzaffar) [reply dmwmbld/gh-bot]: looks like a typo here, looks like a missing `)` at the end (before `]`)?

**Comment by smuzaffar** [comment]
@arooshap , I would suggest to squash the commits too

**Comment by smuzaffar** [comment]
@arooshap , your last changes do not look right. Looks like you have removed the changes for https://github.com/dmwm/deployment/pull/1285/files#diff-79d958b4993ebb2651593d41498307fabe60c300902fa880c8a2a2c29884cfd8R39-R42 section of the code. Again, squash your commits ( or better to start from master branch, apply/commit your changes and force push to [arooshap:resolve-conflicts](https://github.com/arooshap/deployment/tree/resolve-conflicts) branch)

**Comment by arooshap** [comment]
> @arooshap , your last changes do not look right. Looks like you have removed the changes for https://github.com/dmwm/deployment/pull/1285/files#diff-79d958b4993ebb2651593d41498307fabe60c300902fa880c8a2a2c29884cfd8R39-R42 section of the code. Again, squash your commits ( or better to start from master branch, apply/commit your changes and force push to [arooshap:resolve-conflicts](https://github.com/arooshap/deployment/tree/resolve-conflicts) branch)

@smuzaffar Sorry for that, I was in a meeting and forgot to double-check. Here is an updated PR: https://github.com/dmwm/deployment/pull/1286. Sorry for all the noise.

**Pull request closed without merging**